src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -249,8 +249,9 @@ pub fn compile(
     def: &ast::Item,
     edition: Edition
 ) -> SyntaxExtension {
-    let lhs_nm = ast::Ident::from_str("lhs").gensym();
-    let rhs_nm = ast::Ident::from_str("rhs").gensym();
+    let lhs_nm = ast::Ident::with_empty_ctxt(sym::lhs);
+    let rhs_nm = ast::Ident::with_empty_ctxt(sym::rhs);
+    let tt_spec = ast::Ident::with_empty_ctxt(sym::tt);
```

Comment:
```suggestion
    let lhs_nm = ast::Ident::new(sym::lhs, def.span);
    let rhs_nm = ast::Ident::new(sym::rhs, def.span);
    let tt_spec = ast::Ident::new(sym::tt, def.span);
```
Not sure if this span is ever used by diagnostics, but if we have a good span why not use it.